controllers/bounce_processes.go

```diff
@@ -170,11 +170,23 @@ func getProcessesReadyForRestart(logger logr.Logger, cluster *fdbv1beta2.Foundat
 
 	filterConditions := restarts.GetFilterConditions(cluster)
 	var missingProcesses int
+	var markedForRemoval int
 	for _, processGroup := range cluster.Status.ProcessGroups {
-		if cluster.SkipProcessGroup(processGroup) || processGroup.IsMarkedForRemoval() {
+		// Skip process groups that are stuck in terminating. Such a case could represent a kubelet in an unavailable/stuck
+		// state.
+		if processGroup.GetConditionTime(fdbv1beta2.ResourcesTerminating) != nil {
 			continue
 		}
 
+		// We have to count the marked for removal processes in addition to the other process groups. The reason for this
+		// is that a process groups that is marked for removal will require the operator to spin up an additional process group.
+		// This means as long as we have at least one process group that is marked for removal we have more process groups
+		// than the cluster.GetProcessCountsWithDefaults() will return. The total number of process groups will be the result
+		// of cluster.GetProcessCountsWithDefaults() + the number of process groups marked for removal.
+		if processGroup.IsMarkedForRemoval() {
+			markedForRemoval++
+		}
```

Comment:
Aren't these processes already included in "counts.Total()"? It appears not. I am trying to understand why. Maybe you can explain that in the comment above. Thanks!

Reply:
Those are not included as the `counts.Total()` reflects the desired number of processes not the actual number e.g. during a replacement we can have more than the desired number of processes as we spin up a new process for the one that should be replaced and the old one will be there until it's not serving any roles (fully excluded). I'll add this information to the comment 👍

Reply:
Ok, I get the logic now (thanks for your comments about it on Slack). However, this part is still not clear to me:

> The reason for this
>  		// is that a process groups that is marked for removal will require the operator to spin up an additional process group.

These processes are marked for removal (and they are in the process of getting removed, I think) so why would the operator need to start additional processes for these?

Reply:
> The operator has to create a new process in order to match the desired process count,

And we know that the additional processes haven't yet been created (at this point)? Maybe we can discuss this offline, and you can go ahead and merge this PR.